keras_contrib/layers/crf.py

```diff
@@ -282,63 +289,19 @@ def get_config(self):
 
     @property
     def loss_function(self):
-        if self.learn_mode == 'join':
-            def loss(y_true, y_pred):
-                assert self._inbound_nodes, 'CRF has not connected to any layer.'
-                assert not self._outbound_nodes, 'When learn_model="join", CRF must be the last layer.'
-                if self.sparse_target:
-                    y_true = K.one_hot(K.cast(y_true[:, :, 0], 'int32'), self.units)
-                X = self._inbound_nodes[0].input_tensors[0]
-                mask = self._inbound_nodes[0].input_masks[0]
-                nloglik = self.get_negative_log_likelihood(y_true, X, mask)
-                return nloglik
-            return loss
-        else:
-            if self.sparse_target:
-                return sparse_categorical_crossentropy
-            else:
-                return categorical_crossentropy
+        raise AttributeError('"CRF.loss_function" is depreciated. Please use "losses.crf_loss".')
 
     @property
     def accuracy(self):
-        if self.test_mode == 'viterbi':
-            return self.viterbi_acc
-        else:
-            return self.marginal_acc
-
-    @staticmethod
-    def _get_accuracy(y_true, y_pred, mask, sparse_target=False):
-        y_pred = K.argmax(y_pred, -1)
-        if sparse_target:
-            y_true = K.cast(y_true[:, :, 0], K.dtype(y_pred))
-        else:
-            y_true = K.argmax(y_true, -1)
-        judge = K.cast(K.equal(y_pred, y_true), K.floatx())
-        if mask is None:
-            return K.mean(judge)
-        else:
-            mask = K.cast(mask, K.floatx())
-            return K.sum(judge * mask) / K.sum(mask)
+        raise AttributeError('"CRF.accuracy" is depreciated. Please use "metrics.crf_accuracy".')
 
     @property
     def viterbi_acc(self):
-        def acc(y_true, y_pred):
-            X = self._inbound_nodes[0].input_tensors[0]
-            mask = self._inbound_nodes[0].input_masks[0]
-            y_pred = self.viterbi_decoding(X, mask)
-            return self._get_accuracy(y_true, y_pred, mask, self.sparse_target)
-        acc.func_name = 'viterbi_acc'
-        return acc
+        raise AttributeError('"CRF.viterbi_acc" is depreciated. Please use "metrics.crf_viterbi_accuracy".')
```

Comment:
Similar to above. Can this be inferred and the correct value be returned here itself instead of raising an Error?